docs/src/tutorials/tutorial_4.md

```diff
@@ -0,0 +1,304 @@
+- a moving mouth
+- he should say something
+
+This should show you the power of `SubAction` so we want each of those appear (fade in) one after another.
+
+```julia
+using Javis
+
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+function title(args...)
+    fontsize(20)
+    text("Our Mascot", Point(0, -200),
+        valign=:middle, halign=:center)
+end
+
+function face()
+    video = Video(500, 500)
+    javis(video, [
+        BackgroundAction(1:150, ground),
+        Action(title; subactions=[
+            SubAction(1:5, appear(:fade)),
+        ]),
+    ], pathname="jarvis.gif", framerate=15)
+end
+```
+
+The `ground` function should be familiar to you as well as the general structure of the `javis` function.
+
+A new part is the second `Action` which takes in a function and `subactions` as a keyword argument. 
+
+If you're wondering where the `frames` are specified:
+- You can leave the frames blank as here and then the frames from the previous action are used. 
+
+The `subactions` keyword uses a list of [`SubAction`](@ref) structs which are defined in a similar fashion with `frames` and a function but are in some sense simpler than the `Action`. 
+
+A function of a `SubAction` is normally either [`appear`](@ref) or [`disappear`](@ref) at the moment or one of two transformations: [`Translation`](@ref) and [`Rotation`](@ref).
+
+In theory you can define your own but that is way outside of this tutorial.
+
+For the title we want that the text appears in the first 5 frames of the action. More precisely we let it fade in. 
+
+Let's continue with a bit more before we draw part of the mascot.
+
+The following actions will just be added to the `javis` function.
+
+```julia
+Action(16:150, (args...)->circle(O, 100, :stroke); subactions=[
+    SubAction(1:15, appear(:fade)),
+]),
+```
+
+This is very similar to the previous action. Here we can see that `SubAction` uses relative frame numbers such that the head appears in the frames `16:30` and then is at full opacity afterwards.
+
+Just a small refresher: We need the anonymous function `(args...)->circle(O, 100, :stroke)` as each function gets called with the three arguments `video, action, frame`.
+
+Okay let's add some hair shall we?
+
+I want to have some randomness in his hair so let's define:
+
+```julia
+hair_angle = rand(-0.9:0.1:0.9, 20)
+```
+
+at the beginning of the `face` function.
+
+and have a hair function:
+
+```julia
+function hair_blob(angle)
+    sethue("brown")
+    rotate(angle)
+    circle(Point(0, -100), 20, :fill)
+end
+```
+
+It draws one brown hair blob given the angle. We basically rotate the whole canvas and then draw the circle always at the same local position. 
+
+Now how do we draw the hair now without creating an action for each blob?
+
+We can use splatting for that :wink:
+
+```julia
+[
+    Action(26:150, (args...)->hair_blob(hair_angle[i]); subactions=[
+        SubAction(1:25, appear(:fade)),
+    ]) for i=1:20
+]...,
+```
+
```

Comment:
I really like this! :smile: However, you may want to explain what you are doing here exactly with splatting in this function.

Reply:
Hope it's more clear now

Reply:
Perfect! Thanks Ole!